src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts

```diff
@@ -93,17 +93,20 @@ export abstract class AbstractFileDialogService implements IFileDialogService {
 	}
 
 	async preferredHome(schemeFilter = this.getSchemeFilterForWindow()): Promise<URI> {
-
-		// Seek a user-local or user-remote machine-scoped override path string depending on whether caller wants a local or a remote home
-		const inspectedValue = this.configurationService.inspect<string>('files.dialog.defaultPath');
-		const dialogHomePath = schemeFilter === Schemas.file ? inspectedValue.userLocalValue : inspectedValue.userRemoteValue;
-		const userHomePromise = this.pathService.userHome({ preferLocal: schemeFilter === Schemas.file });
-		if (!dialogHomePath) {
-			return userHomePromise;
+		const preferredHomeConfig = this.configurationService.inspect<string>('files.dialog.defaultPath');
+		const preferredHomeCandidate = schemeFilter === Schemas.file ? preferredHomeConfig.userLocalValue : preferredHomeConfig.userRemoteValue;
+		if (preferredHomeCandidate) {
+			const pathLib = await this.pathService.path;
+			if (pathLib.isAbsolute(preferredHomeCandidate)) {
+				const preferredHomeNormalized = pathLib.normalize(preferredHomeCandidate);
+				const preferredHome = resources.toLocalResource(await this.pathService.fileURI(preferredHomeNormalized), this.environmentService.remoteAuthority, this.pathService.defaultUriScheme);
+				if (await this.fileService.exists(preferredHome)) {
```

Comment:
@alexr00 I assume this method `preferredHome` is not called in a loop with hundreds of items right? Because this is a file service call that can take time to resolve.